frontend/src/pages/eventoverview.js

```diff
@@ -0,0 +1,216 @@
+import React from 'react'
+import Router from 'next/router'
+import {
+  Table,
+  Row,
+  Button,
+  Col,
+  Form,
+  FormGroup,
+  Label,
+  Input,
+  Container,
+  Modal,
+  ModalHeader,
+  ModalBody,
+  ModalFooter,
+  Alert
+} from 'reactstrap'
+import ActionButton from '../components/actionButton'
+import Nav from '../components/nav'
+import { getAllEvents, createEvent } from '../utils/api'
+
+class EventOverview extends React.Component<Props> {
+  constructor(props) {
+    super(props)
+    this.state = {
+      events: [],
+      modal: false,
+      name: '',
+      date: '',
+      startTime: '',
+      endTime: '',
+      location: '',
+      description: '',
+      fbLink: '',
+      alert: ''
+    }
+  }
+
+  async componentDidMount() {
+    const events = await getAllEvents()
+    if (events) {
+      this.setState({
+        events: events.result
+      })
+    }
+  }
+
+  toggleModal = () => {
+    this.setState({
+      modal: !this.state.modal
+    })
+  }
+
+  handleChange = e => {
+    this.setState({
+      [e.target.name]: e.target.value
+    })
+  }
+
+  handleKeyPress = async event => {
+    if (event.key === 'Enter') {
+      await this.addEvent()
+    }
+  }
+
+  addEvent = async () => {
+    const event = {
+      name: this.state.name,
+      date: this.state.date,
+      startTime: this.state.startTime,
+      endTime: this.state.endTime,
+      location: this.state.location,
+      description: this.state.description,
+      fbLink: this.state.fbLink
+    }
+    const { code } = await createEvent(event)
+    if (code === 500) {
+      this.setState({
+        alert: 'All fields are required.'
+      })
+    } else {
+      Router.push({ pathname: '/eventoverview' })
+    }
+  }
+
+  render() {
+    let alert
+    if (this.state.alert) {
+      alert = <Alert color="danger">{this.state.alert}</Alert>
+    }
+    return (
+      <>
+        <Nav />
+        <div className="page-content-wrapper">
+          <Container fluid>
+            <Modal isOpen={this.state.modal}>
+              <ModalHeader>Add New Event</ModalHeader>
+              <ModalBody>
+                {alert}
+                <Form>
```

Comment:
If you say `<Form onSubmit={this.addEvent}>` and add `type="submit"` to the submit button below, it should allow you to submit the form by hitting enter.